src/target/riscv/batch.c

```diff
@@ -80,7 +84,7 @@ void riscv_batch_free(struct riscv_batch *batch)
 
 bool riscv_batch_full(struct riscv_batch *batch)
 {
-	return batch->used_scans > (batch->allocated_scans - 4);
+	return batch->used_scans >= (batch->allocated_scans - BATCH_RESERVED_SCANS);
```

Comment:
I would suggest reusing `riscv_batch_available_scans()` here:
```suggestion
	return riscv_batch_available_scans(batch) != 0;
```

Reply:
Thanks for the suggestion - done.

(I think there's a typo and you meant `riscv_batch_available_scans(batch) == 0`, correct? ;-))

Reply:
> (I think there's a typo and you meant `riscv_batch_available_scans(batch) == 0`, correct? ;-))

Oh... Yes, sorry)